src/benchmarks/micro/libraries/System.Collections/Add/AddGivenSize.cs

```diff
@@ -133,5 +134,71 @@ public ObservableCollection<T> ObservableCollection()
             }
             return collection;
         }
+
+        [Benchmark]
+        public ImmutableArray<T> ImmutableArray()
```

Comment:
These benchmarks overall look good, but we should move them out of this class. This class is called Add**GivenSize** because it tests adding new items to collections that support specifying initial capacity. Like `new List<T>(Size)`. We had benchmarks for `AddDefaultSize` (added in https://github.com/dotnet/performance/pull/92), but we have later decided to redesign them to `CreateAddAndClear` (https://github.com/dotnet/performance/pull/663) so they can also cover `Clear`. Please move the new benchmarks from this type to `CreateAddAndClear` and add a `Clear` method invocation to them.